lib/crowbar/init/application.rb

```diff
@@ -424,21 +434,29 @@ class Application < Sinatra::Base
         }
         http_code = 200
 
-        begin
-          test_db_connection(attributes[:postgresql])
-        rescue PG::ConnectionBad => e
-          halt 406, {}, e.message
-        end
 
         result = {}.tap do |res|
+          begin
+            test_db_connection(attributes[:postgresql])
+          rescue PG::ConnectionBad => e
+            res[:errors] = {
+              data: e.message,
+              help: "Make sure connection data is correct and database is reachable"
```

Comment:
I think here a default message to refer to the error message would be enough, as all the details are provided in `e.message`, so i'd just refer to the error message instead of giving this default advice.